shared-internal/src/main/kotlin/io/github/typesafegithub/workflows/shared/internal/app/JwtGenerator.kt

```diff
@@ -0,0 +1,34 @@
+package io.github.typesafegithub.workflows.shared.internal.app
+
+import com.auth0.jwt.JWT
+import com.auth0.jwt.algorithms.Algorithm
+import java.time.Instant
+
+/**
+ * JWT generator that creates JSON Web Tokens (JWT) signed with RSA256.
+ *
+ * This class signs tokens using a provided RSA [privateKey].
+ *
+ * @param githubClientId the public client ID used as the JWT issuer.
+ * @param privateKey the RSA private key file used to sign tokens (expected to be in PKCS#8 format).
+ */
+class JwtGenerator(
+    val githubClientId: String = "Iv23liIZ17VJKUpjacBs", // Public Client ID
+    val privateKey: String = System.getenv("PRIVATE_KEY"),
```

Comment:
Before we go further (and I add the secret), can we make the env var's name a bit more descriptive? I can imagine multiple private keys stored in the future. E. g.
```suggestion
    val privateKey: String = System.getenv("GITHUB_APP_PRIVATE_KEY"),
```

Reply:
Done